### ISSUE TITLE
Unable to Cancel Stock Reconciliation due to e-Invoice fields

### ISSUE THREAD

**Issue by jigneshpshah** [frappe/erpnext#24774]
## Description of the issue

Unable to Cancel Stock Reconciliation due to e-Invoice field validation

## Context information (for bug reports)

**Output of `bench version`**
Installed Apps

ERPNext: v12.18.0 
Frappe Framework: v12.15.0 

## Steps to reproduce the issue

1. Create a Stock Reconciliation
2. Cancel the submitted Stock Reconciliation
3. Below error is generated

### Observed result
![image](https://user-images.githubusercontent.com/4900815/109490580-6ad25700-7aae-11eb-978f-7b706564f215.png)

### Expected result

Should be able to Cancel Stock Reconciliation 

### Stacktrace / full error message

```
raceback (most recent call last):
  File "/home/frappe/frappe-bench/apps/frappe/frappe/desk/form/save.py", line 37, in cancel
    doc.cancel()
  File "/home/frappe/frappe-bench/apps/erpnext/erpnext/stock/doctype/stock_reconciliation/stock_reconciliation.py", line 475, in cancel
    self._cancel()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/model/document.py", line 862, in _cancel
    self.save()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/model/document.py", line 273, in save
    return self._save(*args, **kwargs)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/model/document.py", line 309, in _save
    self.run_before_save_methods()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/model/document.py", line 903, in run_before_save_methods
    self.run_method("before_cancel")
  File "/home/frappe/frappe-bench/apps/frappe/frappe/model/document.py", line 797, in run_method
    out = Document.hook(fn)(self, *args, **kwargs)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/model/document.py", line 1073, in composer
    return composed(self, method, *args, **kwargs)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/model/document.py", line 1056, in runner
    add_to_return_value(self, fn(self, *args, **kwargs))
  File "/home/frappe/frappe-bench/apps/frappe/frappe/model/document.py", line 791, in <lambda>
    fn = lambda self, *args, **kwargs: getattr(self, method)(*args, **kwargs)
  File "/home/frappe/frappe-bench/apps/erpnext/erpnext/controllers/accounts_controller.py", line 118, in before_cancel
    validate_einvoice_fields(self)
  File "/home/frappe/frappe-bench/apps/erpnext/erpnext/__init__.py", line 129, in caller
    return frappe.get_attr(regional_overrides[region][fn_name])(*args, **kwargs)
  File "/home/frappe/frappe-bench/apps/erpnext/erpnext/regional/india/e_invoice/utils.py", line 26, in validate_einvoice_fields
    no_taxes_applied = len(doc.get('taxes', [])) == 0
TypeError: object of type 'NoneType' has no len()
```

**Comment by ankush** [comment]
Closed by #24617

**Comment by jigneshpshah** [comment]
@ankush  , any idea even #24617 got merged on 12th Feb to version-12 hotfix , still it was not released in version-12.18 on 19th Feb ?

We are on frappe.cloud so request you to help (ensure) it  gets merged in version-12 **asap**, as all our stock entries are halted, as we are unable to cancel the incorrect stock reconciliation made by us and we have March year end closing.

Highly appreciate your timely help.

**Comment by jigneshpshah** [comment]
hi @AfshanKhan , 

could you please request @nabinhait , @saurabh6790 to get this vital [minor fix](#24617) #24617 getting merged in version-12, this is a small change but will save the whole financial year end **stock reconciliation** process.

Very Very Important, crucial for all business.

Thanks in advance to all for your kind help. 🙏

**Comment by AfshanKhan** [comment]
Hi, @jigneshpshah this will be part of the next release.
